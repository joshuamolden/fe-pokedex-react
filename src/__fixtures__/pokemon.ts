export const pokemonListMock: PaginatedPokemonList = {
  content: [
    {
      id: 1,
      name: "Bulbasaur",
      types: ["poison", "grass"],
    },
    {
      id: 2,
      name: "Ivysaur",
      types: ["poison", "grass"],
    },
    {
      id: 3,
      name: "Venusaur",
      types: ["poison", "grass"],
    },
    {
      id: 4,
      name: "Charmander",
      types: ["fire"],
    },
    {
      id: 5,
      name: "Charmeleon",
      types: ["fire"],
    },
    {
      id: 6,
      name: "Charizard",
      types: ["fire", "flying"],
    },
    {
      id: 7,
      name: "Squirtle",
      types: ["water"],
    },
    {
      id: 8,
      name: "Wartortle",
      types: ["water"],
    },
    {
      id: 9,
      name: "Blastoise",
      types: ["water"],
    },
    {
      id: 10,
      name: "Caterpie",
      types: ["bug"],
    },
    {
      id: 11,
      name: "Metapod",
      types: ["bug"],
    },
    {
      id: 12,
      name: "Butterfree",
      types: ["flying", "bug"],
    },
    {
      id: 13,
      name: "Weedle",
      types: ["poison", "bug"],
    },
    {
      id: 14,
      name: "Kakuna",
      types: ["poison", "bug"],
    },
    {
      id: 15,
      name: "Beedrill",
      types: ["poison", "bug"],
    },
  ],
  pageable: {
    sort: {
      empty: false,
      sorted: true,
      unsorted: false,
    },
    offset: 0,
    pageNumber: 0,
    pageSize: 15,
    paged: true,
    unpaged: false,
  },
  last: false,
  totalPages: 37,
  totalElements: 553,
  size: 15,
  number: 0,
  sort: {
    empty: false,
    sorted: true,
    unsorted: false,
  },
  numberOfElements: 15,
  first: true,
  empty: false,
};

export const pokemonDetailsMock: PokemonDetails = {
  id: 1,
  name: "Bulbasaur",
  types: ["poison", "grass"],
  height: 7.0,
  weight: 69.0,
  abilities: ["chlorophyll", "overgrow"],
  egg_groups: ["plant", "monster"],
  stats: {
    hp: 45,
    speed: 45,
    attack: 49,
    defense: 49,
    "special-attack": 65,
    "special-defense": 65,
  },
  genus: "Seed Pokémon",
  description:
    "Bulbasaur can be seen napping in bright sunlight.\nThere is a seed on its back. By soaking up the sun’s rays,\nthe seed grows progressively larger.",
};
