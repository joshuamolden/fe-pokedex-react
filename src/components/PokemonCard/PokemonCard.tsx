import React, { ReactElement } from "react";
import "../../pokemonTypes.css";
import { Image, Box, Flex, Text } from "@chakra-ui/react";

const PokemonCard = ({ pokemon }: { pokemon: Pokemon }): ReactElement => (
  <Flex
    w="350px"
    h="400px"
    flexDir="column"
    m="0 auto"
    borderRadius=".2em"
    bgColor="white"
    title="container"
  >
    <Text
      color="black"
      justifyContent="center"
      m="20px"
      fontSize="20px"
      fontWeight="bold"
      title="name"
    >
      {pokemon.name}
    </Text>
    <Box as="hr" h="1px" color="#bababa" title="hr"></Box>
    <Image
      src={`pokemon_imgs/${pokemon.id}.png`}
      alt={pokemon.name}
      display="flex"
      w="250px"
      h="250px"
      m="auto"
      title="image"
    />
    <Flex
      flexDir="row"
      justifyContent="flex-end"
      m="5px"
      mr="10px"
      p="10px"
      title="types"
    >
      {pokemon.types?.map((type, index) => (
        <Flex
          display="flex"
          fontSize="15px"
          mr="10px"
          borderStyle="solid"
          borderRadius="8px"
          paddingLeft="7px"
          paddingRight="7px"
          paddingTop="2px"
          paddingBottom="2px"
          className={type}
          key={index}
        >
          {type.toUpperCase()}
        </Flex>
      ))}
    </Flex>
  </Flex>
);
export default PokemonCard;
