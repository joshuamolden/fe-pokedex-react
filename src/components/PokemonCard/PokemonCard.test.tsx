/* eslint-disable @typescript-eslint/no-empty-function */
import PokemonCard from "./PokemonCard";
import { render } from "@testing-library/react";
import React from "react";
import { pokemonListMock } from "../../__fixtures__/pokemon";

it("render pokemon", () => {
  const { queryByTitle, queryByText } = render(
    <PokemonCard pokemon={pokemonListMock.content[0]} />
  );

  const container = queryByTitle("container");
  const name = queryByText("Bulbasaur");
  const hr = queryByTitle("hr");
  const image = queryByTitle("image");
  const types = queryByTitle("types");
  const poison = queryByText("POISON");
  const grass = queryByText("GRASS");

  expect(container).toBeTruthy();
  expect(name).toBeTruthy();
  expect(hr).toBeTruthy();
  expect(image).toBeTruthy();
  expect(types).toBeTruthy();
  expect(poison).toBeTruthy();
  expect(grass).toBeTruthy();
});

describe("Pokemon.tsx", () => {
  it("render pokemon", () => {
    const { asFragment } = render(
      <PokemonCard pokemon={pokemonListMock.content[0]} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
