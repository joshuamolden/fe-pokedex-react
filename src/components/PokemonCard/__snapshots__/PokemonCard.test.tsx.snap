// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon.tsx render pokemon 1`] = `
<DocumentFragment>
  <div
    class="css-1fk7f3f"
    title="container"
  >
    <p
      class="chakra-text css-1rlrdix"
      title="name"
    >
      Bulbasaur
    </p>
    <hr
      class="css-kuw86o"
      title="hr"
    />
    <img
      alt="Bulbasaur"
      class="chakra-image css-sohjba"
      src="pokemon_imgs/1.png"
      title="image"
    />
    <div
      class="css-ck8nqu"
      title="types"
    >
      <div
        class="poison css-cubg0o"
      >
        POISON
      </div>
      <div
        class="grass css-cubg0o"
      >
        GRASS
      </div>
    </div>
  </div>
</DocumentFragment>
`;
